code/modules/detective_work/evidence.dm

```diff
@@ -80,3 +80,177 @@
+	to_chat(user, "<span class='notice'>You are moving \the [supplied] to \the [src].</span>")
+	return TRUE
+
+/obj/item/sample/print/merge_evidence(obj/item/sample/supplied, mob/user)
+	if(!supplied.evidence || !supplied.evidence.len)
+		return FALSE
+	for(var/print in supplied.evidence)
+		if(evidence[print])
+			evidence[print] = stringmerge(evidence[print],supplied.evidence[print])
+		else
+			evidence[print] = supplied.evidence[print]
+	name = ("[initial(name)] (combined)")
+	to_chat(user, "<span class='notice'>You overlay \the [src] and \the [supplied], combining the print records.</span>")
+	return TRUE
+
+/obj/item/sample/pre_attack(atom/A, mob/living/user, params)
+	. = ..()
+	// Fingerprints will be handled in after_attack() to not mess up the samples taken
+	return A.attackby(src, user, params)
+
+/obj/item/sample/attackby(obj/O, mob/user)
+	if(O.type == src.type)
+		user.unEquip(O)
+		if(merge_evidence(O, user))
+			qdel(O)
+		return TRUE
+	return ..()
+
+/obj/item/sample/fibers
+	name = "\improper fiber bag"
+	desc = "Used to store fiber evidence for forensic examination."
+	icon_state = "fiberbag"
+
+/obj/item/sample/print
+	name = "\improper fingerprint card"
+	desc = "Preserves fingerprints."
+	icon = 'icons/obj/card.dmi'
+	icon_state = "fingerprint0"
+	item_state = "paper"
+
+/obj/item/sample/print/attack_self(mob/user)
+	if(evidence && evidence.len)
+		return
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/H = user
+	if(H.gloves)
+		to_chat(user, "<span class='warning'>Take [H.gloves] off first.</span>")
+		return
+
+	to_chat(user, "<span class='notice'>You press your fingertips firmly against the card.</span>")
+	var/fullprint = H.get_full_print()
+	evidence[fullprint] = fullprint
+	name = ("[initial(name)] ([H])")
+	icon_state = "fingerprint1"
+
+/obj/item/sample/print/attack(mob/living/M, mob/user)
+
+	if(!ishuman(M))
+		return ..()
+
+	if(evidence && evidence.len)
+		return FALSE
+
+	var/mob/living/carbon/human/H = M
+
+	if(H.gloves)
+		to_chat(user, "<span class='warning'>[H] is wearing gloves.</span>")
+		return TRUE
+
+	if(user != H && H.a_intent != INTENT_HELP && !IS_HORIZONTAL(H))
+		user.visible_message("<span class='danger'>[user] tried to fingerprint [H], but he resists.</span>")
+		return TRUE
+
+	if(user.zone_selected == "r_hand" || user.zone_selected == "l_hand")
+		var/has_hand
+		var/obj/item/organ/external/O = H.has_organ("r_hand")
+		if(istype(O))
+			has_hand = TRUE
+		else
+			O = H.has_organ("l_hand")
+			if(istype(O))
+				has_hand = TRUE
+		if(!has_hand)
+			to_chat(user, "<span class='warning'>But [H] has no hands.</span>")
+			return FALSE
+		if(!do_after(user, 2 SECONDS, target = user))
+			return FALSE
+
+		user.visible_message("<span class='notice'>[user] makes a copy of [H]'s fingerprints'.</span>")
+		var/fullprint = H.get_full_print()
+		evidence[fullprint] = fullprint
+		copy_evidence(src)
+		name = ("[initial(name)] (\the [H])")
+		icon_state = "fingerprint1"
```

Comment:
Could you change the icon state in an update_icon_state() override, and then invoke update_appearance() with the right flags here, instead?